src/helpers/sensors/TimeSyncHelper.cpp

```diff
@@ -0,0 +1,139 @@
+#include "TimeSyncHelper.h"
+
+#ifdef ENV_INCLUDE_GPS // remove/extend when time can be synced from other sources
+
+unsigned long TimeSyncHelper::_lastSyncTime = -48 * 60 * 60 * 1000;
+unsigned long TimeSyncHelper::_rolloverCount = 0;
+bool TimeSyncHelper::_gpsWasOnBeforeSync = false;
+bool TimeSyncHelper::_initialized = false;
+bool TimeSyncHelper::_initialSyncCompleted = false;
+
+TimeSyncHelper::SyncState TimeSyncHelper::_syncState = SYNC_IDLE;
+unsigned long TimeSyncHelper::_syncStartTime = 0;
+int TimeSyncHelper::_initialSyncTimeoutCounter = 0;
+
+void TimeSyncHelper::init() {
+    if (_initialized) return;
+    
+    _lastSyncTime = -48 * 60 * 60 * 1000;
+    _rolloverCount = 0;
+    _gpsWasOnBeforeSync = false;
+    _syncState = SYNC_IDLE;
+    _syncStartTime = 0;
+    _initialSyncTimeoutCounter = 0;
+    _initialSyncCompleted = false;
+    _initialized = true;
+    
+    MESH_DEBUG_PRINTLN("TimeSyncHelper initialized");
+}
+
+bool TimeSyncHelper::isInitialSyncCompleted() {
+    return _initialSyncCompleted;
+}
+
+void TimeSyncHelper::process() {
+    if (!_initialized) {
+        init();
+    }
+    
+    if (!_initialSyncCompleted) {
+        if (_initialSyncTimeoutCounter == 0) {
+            _gpsWasOnBeforeSync = sensors.gps_active;
+            if (!sensors.gps_active) {
+                sensors.start_gps();
+            }
+            MESH_DEBUG_PRINTLN("Starting initial GPS sync");
+        }
+
+        _initialSyncTimeoutCounter++;
+
+        if (sensors._location != nullptr) {
```

Comment:
`_location` is only present in `EnvironmentSensorManager` so you make an assumption on the sensor_manager that is here ... could be more generic by using this kind of func to get to the gps (I find it complicated but that's what we have for now): https://github.com/meshcore-dev/MeshCore/blob/95d1f052c21849387db300ab6ab954601e2c51d7/examples/companion_radio/ui-new/UITask.cpp#L750-L770 

I think we could add a more simple getter to access sensors from their name in the code ...